queries/javascript/textobjects.scm

```diff
@@ -38,3 +38,11 @@
 
 ;; blocks
 (_ (statement_block) @block.inner) @block.outer
+
+;; parameters
+(formal_parameters
+  (identifier) @parameter.inner)
+
+;; arguments
+(arguments
+  (identifier) @parameter.inner)
```

Comment:
Maybe also lambda parameters (from highlights):
```
;; Lambda parameter
(inferred_parameters (identifier) @parameter) ; (x,y) -> ...
(lambda_expression
    parameters: (identifier) @parameter) ; x -> ...
```

Reply:
Hmm interesting, I used the playground with some anonymous functions and the current queries worked for those. I'll add the lambda parameters just in case.

While I have your attention, I noticed the current query for arguments doesn't really work with all cases, because an argument isn't always an `identifier`. Example that fails:

```typescript
    return new Promise<RespType>((resolve, reject) => {
      this.client.makeUnaryRequest<ReqType, RespType>(
        `/company.Service/${method}`,
        (req) => Buffer.from(encoder.encode(req).finish()),
        (buf) => (decoder.decode(buf) as unknown) as RespType,
        data,
        this.grpcMeta,
        { deadline },
        (err: grpc.ServiceError | null, resp: RespType | undefined) => {
          if (err) {
            reject(err)
          } else {
            resolve(resp)
          }
        }
      )
    })
```

The first parameter to makeUnaryRequest is actually `template_string`, while the second and third are `arrow_function`. Doing `(arguments (_) @parameter.inner)` fixes it (and every other case I tested), but is that too open?

Reply:
Would be ok for me. How does that handle comments? It's also possible that the JS highlights are outdated/incomplete.

Reply:
Apprently, arguments can have any expression as child.
```
      "types": [
        {
          "type": "_expression",
          "named": true
        },
        {
          "type": "spread_element",
          "named": true
        }
      ]
    }
```
https://github.com/nvim-treesitter/nvim-treesitter/pull/453 would bring us `(not-has-type? @parameter.inner "comment")`

Reply:
@fsouza sorry, those queries were from Java (was fuzzy searching Java script)
these are the ones from JS:
```
(formal_parameters (identifier) @parameter)

(formal_parameters
  (rest_parameter
    (identifier) @parameter))

; ({ a }) => null
(formal_parameters
  (object_pattern
    (shorthand_property_identifier) @parameter))

; ({ a: b }) => null
(formal_parameters
  (object_pattern
    (pair
      value: (identifier) @parameter)))

; ([ a ]) => null
(formal_parameters
  (array_pattern
    (identifier) @parameter))

; a => null
(variable_declarator                                                                                                                                                                                        
    value: (arrow_function                                                                                                                                                                                    
      parameter: (identifier) @parameter))
```

Reply:
So you could also have optional parameters in JS I guess for the highlights.

Reply:
Oh wait it looks like lambda_expression is a java thing, not javascript?

Reply:
Right, I was grepping the wrong language.

Reply:
Made the change to the arguments query: https://github.com/nvim-treesitter/nvim-treesitter-textobjects/pull/19/commits/0ad8731bf7ccda7cb58a3eaa1616d72357966103

Let me know if it looks good, then I can squash it :D

Reply:
Yeah, looks good. Maybe we can tune this not-a-comment behavior later when we have predicate for it.

Reply:
Should be fine :+1:

Reply:
I double check the arguments appear to be fine!